.changeset/renamed_useimportrestrictions_to_nopackageprivateimports.md

```diff
@@ -0,0 +1,8 @@
+---
+cli: major
+---
+
+# The rule `useImportRestrictions` has been renamed to `noPackagePrivateImports`
+
+To avoid confusion with `noRestrictedImports`, `useImportRestrictions` has been
```

Comment:
Since `noRestrictedImports` is a nursery rule, we can make this a patch, because nursery rules don't follow semver